src/QsCompiler/Core/Transformations/ExpressionTransformation.fs

```diff
@@ -17,51 +17,36 @@ open Microsoft.Quantum.QsCompiler.Transformations.Core.Utils
 
 type private ExpressionKind = QsExpressionKind<TypedExpression, Identifier, ResolvedType>
 
+type ExpressionKindTransformationBase(expressionTransformation: _ -> ExpressionTransformationBase, options) =
+    static let createExpressionTransformation kinds options =
+        let types = TypeTransformationBase options
+        let expressions = ExpressionTransformationBase((fun () -> kinds), (fun () -> types), options)
+        fun () -> expressions
```

Comment:
So this is the actual body of the new(options) constructor, right? The reason it needs to be here instead of at the definition of the new(options) constructor is because it needs to be passed into that constructor's call to the default constructor, and F# demands the call to the default constructor happens first, is that right?

Correct me if I'm wrong, but basically we would want
```
new(options: TransformationOptions) as this =
    let typeTransformation = new TypeTransformationBase(options)    
    let expressionTransformation =
        ExpressionTransformationBase((fun _ -> this), (fun _ -> typeTransformation), options)
    ExpressionKindTransformationBase((fun () -> expressionTransformation), options)
```
But F# doesn't allow for the default constructor to be called that late in the definition. Is my understanding of why this static let is necessary correct?

Reply:
Yes. The restriction isn't specific to F#, by the way. C# also requires the alternate constructor to be called first.

Reply:
I just realized that F# doesn't actually have this restriction. I replaced the `static let`s with inline `let`s.

Reply:
Nice catch! That makes things even cleaner looking.